src/Adapter/Hook/HookDispatcher.php

```diff
@@ -56,12 +57,29 @@ class HookDispatcher extends EventDispatcher implements HookDispatcherInterface
      */
     private $requestStack;
 
+    /**
+     * @var HookRegistry
+     */
+    private $hookRegistry;
+
+    /**
+     * @var bool
+     */
+    private $isDebug;
+
     /**
      * @param RequestStack|null $requestStack (nullable to preserve backward compatibility)
+     * @param HookRegistry|null $hookRegistry (nullable to preserve backward compatibility)
+     * @param bool $isDebug
      */
-    public function __construct(RequestStack $requestStack = null)
-    {
+    public function __construct(
+        RequestStack $requestStack = null,
+        HookRegistry $hookRegistry = null,
+        bool $isDebug = false
```

Comment:
How about decorating this service and injecting hookdispatcher in dev mode instead of adding arguments ? This adds a new class but avoids nullable constructors

Reply:
That's a very interesting idea. However, I would need to call `getListeners` again in the decorator so it's a bit less optimized but since it's only in dev mode maybe it's acceptable and it should be quick anyway